osgar/drivers/kloubak.py

```diff
@@ -430,31 +609,48 @@ def update_drive_front_wheels(self, timestamp, data):
-                self.publish('can', CAN_triplet(0x13, stop))  # right rear
-                self.publish('can', CAN_triplet(0x14, stop))  # left rear
+                self.publish('can', CAN_triplet(0x13, stop))  # right rear/mid
+                self.publish('can', CAN_triplet(0x14, stop))  # left rear/mid
+                
+                self.publish('can', CAN_triplet(0x15, stop))  # back right
+                self.publish('can', CAN_triplet(0x16, stop))  # back left
 
             elif self.desired_speed < 0:
                 if self.last_encoders_rear_right is not None:
-                    self.publish('can', CAN_triplet(0x33, list(struct.pack('>i', limit_r))))
+                    self.publish('can', CAN_triplet(0x35, list(struct.pack('>i', limit_r))))  # self.publish('can', CAN_triplet(0x33, list(struct.pack('>i', limit_r))))
                 if self.last_encoders_rear_left is not None:
-                    self.publish('can', CAN_triplet(0x34, list(struct.pack('>i', limit_l))))
+                    self.publish('can', CAN_triplet(0x36, list(struct.pack('>i', limit_l))))  # self.publish('can', CAN_triplet(0x34, list(struct.pack('>i', limit_l))))
                 self.publish('can', CAN_triplet(0x11, stop))  # right front
                 self.publish('can', CAN_triplet(0x12, stop))  # left front 
+                
+                self.publish('can', CAN_triplet(0x13, stop))  # rear right/mid
+                self.publish('can', CAN_triplet(0x14, stop))  # rear left/mid 
 
             else:
                 self.publish('can', CAN_triplet(0x11, stop))  # right front
                 self.publish('can', CAN_triplet(0x12, stop))  # left front
-                self.publish('can', CAN_triplet(0x13, stop))  # right rear
-                self.publish('can', CAN_triplet(0x14, stop))  # left rear
+                self.publish('can', CAN_triplet(0x13, stop))  # right rear/mid
+                self.publish('can', CAN_triplet(0x14, stop))  # left rear/mid
+                
+                self.publish('can', CAN_triplet(0x15, stop))  # back right
+                self.publish('can', CAN_triplet(0x16, stop))  # back left
 
     def slot_can(self, timestamp, data):
         self.time = timestamp
-        if self.drive_mode == DriveMode.ALL:
-            self.update_drive_four_wheels(timestamp, data)
-        elif self.drive_mode == DriveMode.FRONT:
-            self.update_drive_front_wheels(timestamp, data)
+        if self.num_axis == 3:
+            self.update_drive_three_axles(timestamp, data)
+
+        elif self.num_axis == 2:
+            if self.drive_mode == DriveMode.ALL:
+                self.update_drive_two_axles(timestamp, data)
+            elif self.drive_mode == DriveMode.FRONT:
+                self.update_drive_front_wheels(timestamp, data)
+            else:
+                assert False, self.drive_mode  # is not supported yet
+
         else:
-            assert False, self.drive_mode  # is not supported yet
```

Comment:
this assert should remain in the original if-else for `self.drive_mode`

Reply:
I think that the assert you mean is still there (line 649).